src/shared/watchdog.c

```diff
@@ -59,31 +72,13 @@ static int settimeout_watchdog(void) {
         sec = MIN(t, (usec_t) INT_MAX); /* Saturate */
 
         r = ioctl(watchdog_fd, WDIOC_SETTIMEOUT, &sec);
-        if (r < 0) {
-                /* Setting the timeout might not be supported by the HW. In
-                 * that case we'll reuse the effective timeout. */
-                if (!ERRNO_IS_NOT_SUPPORTED(errno))
-                        return log_warning_errno(errno, "Failed to set timeout to %is: %m", sec);
-
-                if (ioctl(watchdog_fd, WDIOC_GETTIMEOUT, &sec) < 0)
-                        return log_warning_errno(errno, "Failed to request watchdog HW timeout: %m");
-
-                op_not_supported = true;
-        }
+        if (r < 0)
+                return log_warning_errno(errno, "Failed to set timeout to %is: %m", sec);
```

Comment:
you basically revert the earlier fix here. not following?

the idea is not *accept* and *not* *warn* about devices where WDIOC_SETTIMEOUT fails?

Reply:
right we shouldn't emit a warning in case the driver doesn't support WDIOC_SETTIMEOUT, I'll fix it.